testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/DeleteAccountTest.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.testsuite.ui.account2;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+import org.jboss.arquillian.graphene.page.Page;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.keycloak.events.EventType;
+import org.keycloak.models.AccountRoles;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.representations.idm.RequiredActionProviderRepresentation;
+import org.keycloak.representations.idm.RoleRepresentation;
+import org.keycloak.testsuite.AssertEvents;
+import org.keycloak.testsuite.admin.ApiUtil;
+import org.keycloak.testsuite.ui.account2.page.AIALoginPage;
+import org.keycloak.testsuite.ui.account2.page.AbstractLoggedInPage;
+import org.keycloak.testsuite.ui.account2.page.DeleteAccountActionConfirmPage;
+import org.keycloak.testsuite.ui.account2.page.DeleteAccountPage;
+import org.keycloak.testsuite.ui.account2.page.WelcomeScreen;
+
+/**
+ * @author Zakaria Amine <zakaria.amine88@gmail.com>
+ */
+public class DeleteAccountTest extends BaseAccountPageTest {
+
+  @Page
+  private DeleteAccountPage deleteAccountPage;
+
+  @Page
+  private WelcomeScreen welcomeScreen;
+
+  @Page
+  private AIALoginPage aiaLoginPage;
+
+  @Page
+  private DeleteAccountActionConfirmPage deleteAccountActionConfirmPage;
+
+  @Rule
+  public AssertEvents events = new AssertEvents(this);
+
+  @Override
+  protected AbstractLoggedInPage getAccountPage() {
+    return deleteAccountPage;
+  }
+
+  @Override
+  public void navigateBeforeTest() {
+    getAccountPage().navigateTo();
+    loginToAccount();
+  }
+
+  @Override
+  @Test
+  public void navigationTest() {
+    enableDeleteAccountRequiredAction();
+    addDeleteAccountRoleToUserClientRoles();
+    driver.navigate().refresh();
+    super.navigationTest();
+    disableDeleteAccountRequiredAction();
+    removeDeleteAccountRoleToUserClientRoles();
+  }
+
+  @Test
+  public void deleteOwnAccountPageNotVisibleAndNotAccessibleWithoutUserRole() {
+    enableDeleteAccountRequiredAction();
+   Assert.assertFalse(welcomeScreen.isDeleteAccountLinkVisible());
+    deleteAccountPage.navigateTo();
+    pageNotFound.assertCurrent();
+    //reset role back since realm is shared among tests
+    disableDeleteAccountRequiredAction();
```

Comment:
This should be in a `finally` block or in `@After` method. In case the test fails, the cleanup is not performed. This applies to other test methods too.

Reply:
:+1: